examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -0,0 +1,320 @@
+        # IMPORTANT: use a key that is consistent with kwargs of orchestrator_aggregation()
+        "weights": silo_training_step.outputs.model
+    }
+
+
+def orchestrator_aggregation(weights=[]):
+    """Create steps for running FL training in the silo.
+
+    Args:
+        train_data (Input): preprocessed data (see silo_training())
+        test_data (Input): preprocessed data (see silo_training())
+        running_checkpoint (Input): if not None, the checkpoint obtained from previous iteration (see orchestrator_aggregation())
+        lr (int): learning rate for training component
+        batch_size (int): batch size for training component
+        epochs (int): epochs for training component
+
+    Returns:
+        PipelineStep: the training step of the FL pipeline
+        Dict[str, Input]: a map of the inputs expected as kwargs by orchestrator_aggregation()
+    """
+    # create some custom map from silo_outputs_map to expected inputs
+    aggregation_inputs = dict(
+        [
+            (f"input_silo_{silo_index+1}", silo_output)
+            for silo_index, silo_output in enumerate(weights)
+        ]
+    )
+
+    # aggregate all silo models into one
+    aggregate_weights_step = aggregate_component(**aggregation_inputs)
+
+    return aggregate_weights_step, {
+        # IMPORTANT: use a key that is consistent with kwargs of silo_training()
+        "running_checkpoint": aggregate_weights_step.outputs.aggregated_output
+    }
+
+
+#######################
+### D. FACTORY CODE ###
+#######################
+
+# In this section, we're using a wrapper to build the full FL pipeline
+# based on the custom methods you implemented in section C.
+
+# 1. create an instance of the factory
+builder = FederatedLearningPipelineFactory()
+
+# 2. feed it with FL parameters
+
+builder.set_orchestrator(
+    # provide settings for orchestrator
+    YAML_CONFIG.federated_learning.orchestrator.compute,
+    YAML_CONFIG.federated_learning.orchestrator.datastore,
+)
+
+for silo_config in YAML_CONFIG.federated_learning.silos:
+    builder.add_silo(
+        # provide settings for this silo
+        silo_config.compute,
+        silo_config.datastore,
+        # any additional custom kwarg will be sent to silo_preprocessing() as is
+        raw_train_data=Input(
+            type=silo_config.training_data.type,
+            mode=silo_config.training_data.mode,
+            path=silo_config.training_data.path,
+        ),
+        raw_test_data=Input(
+            type=silo_config.testing_data.type,
+            mode=silo_config.testing_data.mode,
+            path=silo_config.testing_data.path,
+        ),
+    )
+
+# 3. use a pipeline factory method
+
+pipeline_job = builder.build_basic_fl_pipeline(
+    # building requires all 3 functions provided as argument below
+    silo_preprocessing,
+    silo_training,
+    orchestrator_aggregation,
+    # RESERVED: this kwarg is for building iterations
+    iterations=YAML_CONFIG.training_parameters.num_rounds,
```

Comment:
to be consistent and more clearer to users, would that be better to also use "iterations" rather than "num_rounds" in the FL config.yaml file? That is also what we did in shrike FL API, i.e., we always use the term "iteration".